pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3139,7 +3139,7 @@ func buildListener(port v1.ServicePort, annotations map[string]string, sslPorts
 	listener.InstancePort = &instancePort
 	listener.LoadBalancerPort = &loadBalancerPort
 	certID := annotations[ServiceAnnotationLoadBalancerCertificate]
-	if certID != "" && (sslPorts == nil || sslPorts.numbers.Has(loadBalancerPort) || sslPorts.names.Has(portName)) {
+	if certID != "" && (sslPorts == nil || sslPorts.numbers.Has(loadBalancerPort) || sslPorts.names.Has(portName) || len(sslPorts.numbers) > 0) {
```

Comment:
I don't think this is right - today when I have a k8s service that listens on port 80 and on port 443, I can configure `load-balancer-ssl-ports=443`, and then ELB will do https only on port 443, port 80 will listen for http.  The current behaviour will then send non-TLS traffic onto port 443 which is at the very least unintuitive.

But if we add `len(sslPorts.numbers) > 0` here we will break port 80.